src/algorithms/ctmrg.jl

```diff
@@ -1,24 +1,46 @@
+# TODO: add option for different projector styles (half-infinite, full-infinite, etc.)
+"""
+    struct ProjectorAlg{S}(; svd_alg = TensorKit.SVD(), trscheme = TensorKit.notrunc(),
+                           fixedspace = false, verbosity = 0)
+
+Algorithm struct collecting all projector related parameters. The truncation scheme has to be
+a `TensorKit.TruncationScheme`, and some SVD algorithms might have further restrictions on what
+kind of truncation scheme can be used. If `fixedspace` is true, the truncation scheme is set to
+`truncspace(V)` where `V` is the environment bond space, adjusted to the corresponding
+environment direction/unit cell entry.
+"""
+@kwdef struct ProjectorAlg{S}
+    svd_alg::S = TensorKit.SVD()
+    trscheme::TruncationScheme = TensorKit.notrunc()
+    fixedspace::Bool = false
+    verbosity::Int = 0
+end
+
+function fix_spaces(alg::ProjectorAlg)
+    return ProjectorAlg(;
+        svd_alg=alg.svd_alg, trscheme=alg.trscheme, fixedspace=true, verbosity=alg.verbosity
+    )
+end
+
 # TODO: add abstract Algorithm type?
 """
-    struct CTMRG(; trscheme = TensorKit.notrunc(), tol = Defaults.ctmrg_tol,
-                 maxiter = Defaults.ctmrg_maxiter, miniter = Defaults.ctmrg_miniter,
-                 verbosity = 0, fixedspace = false)
+    struct CTMRG(; tol = Defaults.ctmrg_tol, maxiter = Defaults.ctmrg_maxiter,
+                 miniter = Defaults.ctmrg_miniter, verbosity = 0,
+                 projector_alg = ProjectorAlg())
 
 Algorithm struct that represents the CTMRG algorithm for contracting infinite PEPS.
-The projector bond dimensions are set via `trscheme` which controls the truncation
-properties inside of `TensorKit.tsvd`. Each CTMRG run is converged up to `tol`
-where the singular value convergence of the corners as well as the norm is checked.
-The maximal and minimal number of CTMRG iterations is set with `maxiter` and `miniter`.
-Different levels of output information are printed depending on `verbosity` (0, 1 or 2).
-Regardless of the truncation scheme, the space can be kept fixed with `fixedspace`.
+Each CTMRG run is converged up to `tol` where the singular value convergence of the
+corners as well as the norm is checked. The maximal and minimal number of CTMRG iterations
+is set with `maxiter` and `miniter`. Different levels of output information are printed
+depending on `verbosity` (0, 1 or 2). All projector related properties are set using the
+`ProjectorAlg` struct.
 """
 @kwdef struct CTMRG
-    trscheme::TruncationScheme = TensorKit.notrunc()
     tol::Float64 = Defaults.ctmrg_tol
     maxiter::Int = Defaults.ctmrg_maxiter
     miniter::Int = Defaults.ctmrg_miniter
     verbosity::Int = 0
-    fixedspace::Bool = false
+    projector_alg::ProjectorAlg = ProjectorAlg()
```

Comment:
It would be cool if we could hide this from an end-user. Ie, get a `CTMRG(; kwargs...)` function that takes in some reasonable kwargs (verbosity, trscheme, svdalg...?), and then automatically constructs the projector algorithm.

Reply:
Yes, totally agree, that's much more useful than having to initialize a `ProjectorAlg` every time. I'll implement such a convenience constructor.